enhancements/machine-config/machine-config-irreconcilable-changes.md

```diff
@@ -0,0 +1,444 @@
+apply the requested changes.
+
+MachineConfig daemons will continue to perform the already supported updates 
+to nodes, no matter if irreconcilable validations are skipped or not. Any 
+change to the MachineConfig out of supported ones is ignored.
+
+When the update process successfully finishes the MCD reports by performing 
+two actions:
+
+- Calculate the difference between the irreconcilable fields used at cluster 
+installation time and after the rendered configuration apply. The difference 
+result, as a per-field human readable string, is set in the 
+`irreconcilableChangesDiff` status field of the MCN.
+
+- Update the `machineconfiguration.openshift.io/currentConfig` annotation to
+point to the updated one and `machineconfiguration.openshift.io/state` to 
+`Done`.
+
+Nodes joining the cluster will never have irreconcilable differences between 
+the rendered configuration and the initial Ignition parameters, as both will 
+match. If at some point of the lifespan of a node this feature is enabled 
+and the node observes irreconcilable differences, the 
+`irreconcilableChangesDiff` field will be set in its MCN and will never be
+cleared.
+
+### API Extensions
+
+#### MachineConfiguration
+
+Update the `MachineConfiguration` CRD to add a field that lets the user skip
+individual irreconcilable fields validations called 
+`irreconcilableMCValidationOverrides`.
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: MachineConfiguration
+metadata:
+    name: cluster
+spec:
+  # ... Already existing spec fields
+  irreconcilableMCValidationOverrides:
+    - StorageDisks       # Allow changes to Ignition's disks section
+    - StorageFileSystems # Allow changes to Ignition's filesystem section
+    - StorageRaid        # Allow changes to Ignition's raid  section
+status:
+  # ...
+```
+
+Based on customer use-cases the MCO has encountered over the years, we have 
+determined that supporting irreconcilable changes for partition schema 
+modifications will address the primary operational need driving this enhancement. 
+Customer requests consistently center around deploying nodes with different 
+storage configurations than those used during initial cluster deployment.
+
+This enhancement therefore focuses on supporting 
+`irreconcilableMCValidationOverrides` for the specific fields required to 
+modify partition schemas:
+
+* StorageDisks
+* StorageFileSystems
+* StorageRaid
+
+These three fields provide complete coverage for storage configuration 
+scenarios while maintaining a focused scope that aligns with demonstrated 
+customer needs. Additional fields such as `Passwd`, `StorageDirectories`, 
+`StorageFiles` and `StorageLinks` are not included in this initial 
+implementation as they fall outside the scope of partition schema changes and 
+have not been identified as part of the relevant customer use cases.
+
+The `MachineConfiguration` status section does not require any change.
```

Comment:
Is there any reason why the MCO might reject the request to ignore irreconcilable changes?

Reply:
Ack, was considering whether we should be thinking about having the spec changes mirrored in status to allow the MCO to "accept" the changes, copy to status, and then consumers of these fields read from there, but sounds like it's not required